### PR TITLE
adds colony lathe to the lifebringer vault

### PR DESCRIPTION
**Pull request by Makeshift-coder**
## About The Pull Request

exactly what the title says

## Why It's Good For The Game

give lifebringers the ability to expand a little without a direct access to guns or any other rnd stuff, just stuff like creating seed rooms outside and building things instead of being the old "do hydro until bored" cycle

## Proof Of Testing
- Game compiles
image
- Server starts
image
- map generation loads fine server does not give off any errors
image
- Extras
image
image

## Changelog

:cl:
add: adds an unpacked colony fabricator
add: replaces the four bluespace beakers on a table with a bluespace beaker box for ease of access
del: Removed a single sink to put the unpacked fabricator on it

/:cl:

### CODE REVIEW

**Review by Majkl-J — COMMENTED** [review]
This will most likely get overriden next upstream merge

**Comment by Makeshift-coder** [comment]
this is not a groundbreaking change, it won't require much effort to redo

**Comment by StrangeWeirdKitten** [comment]
You can add it to the automapper and point it to the lifebringer vault areas. The automatter has special lavaland functions.

**Comment by Makeshift-coder** [comment]
headache..

**Comment by Makeshift-coder** [comment]
figuring out what's wrong with the automapper generation is an even worse headache, adding entries either requires something not specifically stated or it's voodoo spessmagic

**Comment by StrangeWeirdKitten** [comment]
> figuring out what's wrong with the automapper generation is an even worse headache, adding entries either requires something not specifically stated or it's voodoo spessmagic

let me figure it out for you

**Comment by Makeshift-coder** [comment]
also automapper isn't meant for map changes, it's meant for map generations iirc, like mapping a randomized always spawning section (centcom) seed vault has a chance of spawning, and for some reasons making it the target of the automapper means it won't work alongisde admin tools like when someone ahelps for the seed vault to be spawned.

**Comment by Makeshift-coder** [comment]
this is what I added.

image

**Comment by StrangeWeirdKitten** [comment]
> also automapper isn't meant for map changes, it's meant for map generations afak, like mapping a randomized always spawning section (centcom) seed vault has a chance of spawning, and for some reasons making it the target of the automapper means it won't work alongisde admin tools like when someone ahelps for the seed vault to be spawned.

There's a varient of the automapper that will randomly place things around the station. For example, the command waredrobe.

**Comment by Makeshift-coder** [comment]
> > also automapper isn't meant for map changes, it's meant for map generations afak, like mapping a randomized always spawning section (centcom) seed vault has a chance of spawning, and for some reasons making it the target of the automapper means it won't work alongisde admin tools like when someone ahelps for the seed vault to be spawned.
> 
> There's a varient of the automapper that will randomly place things around the station. For example, the command waredrobe.

I'm not fully versed in the thing, I'm still just trying to figure out why it's not accepting my entry. Since bubber has a lot of random duplicated files in the code, I can't really know which is which without reading specific code that tells me which one is utilized. At least I could read that the server uses the bubber automapper folder not the skyrat one.

**Comment by StrangeWeirdKitten** [comment]
Make a file inside ``modular_zubbers/code`` and name it "automapper"

Then create a .dm file called ``area_spawn_entries`` and place this code inside it.
```
/datum/area_spawn/podspawn_colony_lathe
	target_areas = list(/area/ruin/powered/seedvault, /area/centcom/central_command_areas/supply)
	desired_atom = /obj/machinery/rnd/production/colony_lathe
	mode = AREA_SPAWN_MODE_HUG_WALL
```

It's a little hacky, but it will spawn the lathe if the seedvault exists. Otherwise, it'll just appear inside the supply centcomm area. The code isn't built for ruins. But this is a workaround

**Comment by Makeshift-coder** [comment]
> Make a folder inside `modular_zubbers/code` and name it "automapper"
> 
> Then create a .dm file called `area_spawn_entries` and place this code inside it.
> 
> ```
> /datum/area_spawn/podspawn_colony_lathe
> 	target_areas = list(/area/ruin/powered/seedvault, /area/centcom/central_command_areas/supply)
> 	desired_atom = /obj/machinery/rnd/production/colony_lathe
> 	mode = AREA_SPAWN_MODE_HUG_WALL
> ```
> 
> It's a little hacky, but it will spawn the lathe if the seedvault exists. Otherwise, it'll just appear inside the supply centcomm area. The code isn't built for ruins. But this is the best workaround to the issue.
> 
> I haven't fully tested this, but if this doesn't work, then sadly it's not possible with current code

there already is `modular_zubbers\code\modules\automapper` within the codebase, should I not use that?

**Comment by StrangeWeirdKitten** [comment]
yeah use that

**Comment by Makeshift-coder** [comment]
it works, for a hacky piece of code, yes

**Comment by Makeshift-coder** [comment]
it works for a hacky piece of code

image

**Comment by StrangeWeirdKitten** [comment]
> it works for a hacky piece of code
> 
> image

guess it decided to spawn on the spawners, lol, but it can always be moved

**Comment by Makeshift-coder** [comment]
im gonna update the pr with this, because I ain't gonna bother blackmagicking the pr to work, I'll delete the automapper things I placed until I can figure them out.

**Comment by Makeshift-coder** [comment]
image

does work after deleting old things I made, and undoes my map changes to the original map. game compiles and runs fine as shown in screenshot of the game starting and spawning the desired object.

File: modular_zubbers/code/modules/automapper/code/area_spawn_entries.dm [hunk 1]
```diff
@@ -1,3 +1,10 @@
 /datum/area_spawn/secmed_landmark
 	target_areas = list(/area/station/security/medical, /area/station/security/brig)
 	desired_atom = /obj/effect/landmark/start/security_medic
+
+
+
+/datum/area_spawn/podspawn_colony_lathe
```
Comment (StrangeWeirdKitten) [inline modular_zubbers/code/modules/automapper/code/area_spawn_entries.dm:7]:
```suggestion

/datum/area_spawn/podspawn_colony_lathe
```
Very minor

**Review by StrangeWeirdKitten — APPROVED** [review]

**Pull request merged**